File: frontend/src/types/businessRule.ts
export type BusinessRuleApprovalStatus =
  | "draft"
  | "generated"
  | "approved"
  | "rejected";

export type BusinessRuleOperator =
  | "="
  | "!="
  | ">"
  | ">="
  | "<"
  | "<="
  | "like"
  | "not like"
  | "between_or_between";

export interface BusinessRuleEntity {
  id: number;
  name: string;
  classification: string;
  approval_status: string;
}

export interface BusinessRuleColumn {
  id: number;
  metadata_table_id: number;
  column_name: string;
  business_name: string | null;
  data_type: string;
  classification: string;
  is_sensitive: boolean;
}

export interface BusinessRule {
  id: number;

  business_entity_id: number;
  metadata_column_id: number;

  name: string;
  trigger_phrase: string;

  synonyms: string | null;

  operator: BusinessRuleOperator;
  rule_value: string;

  description: string | null;

  confidence: number;

  approval_status: BusinessRuleApprovalStatus;
  is_active: boolean;

  created_at: string;
  updated_at: string;

  entity: BusinessRuleEntity;
  column: BusinessRuleColumn;
}

export interface BusinessRuleCreate {
  business_entity_id: number;
  metadata_column_id: number;

  name: string;
  trigger_phrase: string;

  synonyms?: string | null;

  operator: BusinessRuleOperator;
  rule_value: string;

  description?: string | null;

  confidence: number;

  approval_status: BusinessRuleApprovalStatus;
  is_active: boolean;
}

export interface BusinessRuleUpdate {
  business_entity_id?: number;
  metadata_column_id?: number;

  name?: string;
  trigger_phrase?: string;

  synonyms?: string | null;

  operator?: BusinessRuleOperator;
  rule_value?: string;

  description?: string | null;

  confidence?: number;

  approval_status?: BusinessRuleApprovalStatus;
  is_active?: boolean;
}

File: backend/app/schemas/business_rule.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class BusinessRuleCreate(BaseModel):
    business_entity_id: int
    metadata_column_id: int

    name: str
    trigger_phrase: str

    synonyms: str | None = None

    operator: str = "="
    rule_value: str

    description: str | None = None

    confidence: int = 100

    approval_status: str = "draft"
    is_active: bool = True


class BusinessRuleUpdate(BaseModel):
    business_entity_id: int | None = None
    metadata_column_id: int | None = None

    name: str | None = None
    trigger_phrase: str | None = None

    synonyms: str | None = None

    operator: str | None = None
    rule_value: str | None = None

    description: str | None = None

    confidence: int | None = None

    approval_status: str | None = None
    is_active: bool | None = None


class BusinessRuleEntityResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    name: str
    classification: str
    approval_status: str


class BusinessRuleColumnResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    metadata_table_id: int
    column_name: str
    business_name: str | None
    data_type: str
    classification: str
    is_sensitive: bool


class BusinessRuleResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: int

    business_entity_id: int
    metadata_column_id: int

    name: str
    trigger_phrase: str

    synonyms: str | None

    operator: str
    rule_value: str

    description: str | None

    confidence: int

    approval_status: str
    is_active: bool

    created_at: datetime
    updated_at: datetime

    entity: BusinessRuleEntityResponse
    column: BusinessRuleColumnResponse